Source/WebCore/Modules/url-pattern/URLPattern.idl

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2024 Apple Inc.  All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1.  Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ * 2.  Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ * 3.  Neither the name of Apple Inc. ("Apple") nor the names of
+ *     its contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE AND ITS CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL APPLE OR ITS CONTRIBUTORS BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+// https://urlpattern.spec.whatwg.org
+
+typedef (USVString or URLPatternInit) URLPatternInput;
+
+[
+    EnabledBySetting=URLPatternAPIEnabled,
+    Exposed=(Window,Worker)
+] interface URLPattern {
+    constructor(URLPatternInit input, USVString baseURL, optional URLPatternOptions options);
+    constructor(optional URLPatternInput input, optional URLPatternOptions options);
+
+    boolean test(optional URLPatternInput input, optional USVString baseURL);
+  
+    URLPatternResult? exec(optional URLPatternInput input, optional USVString baseURL);
+
+    readonly attribute USVString protocol;
+    readonly attribute USVString username;
+    readonly attribute USVString password;
+    readonly attribute USVString hostname;
+    readonly attribute USVString port;
+    readonly attribute USVString pathname;
+    readonly attribute USVString search;
+    readonly attribute USVString hash;
+
+    readonly attribute boolean hasRegExpGroups;
+};
+
+typedef (USVString or URLPatternInit or URLPattern) URLPatternCompatible;
```

Comment:
Doesn't appear to be used?

Reply:
It was part of the IDL spec so I implemented it. Do you think the spec writers just forgot to remove it?

https://urlpattern.spec.whatwg.org/#typedefdef-urlpatterncompatible

Reply:
They might have, I don't see it being used in the spec either. I would omit for now.